### PR TITLE
Change salesEndDate on event start date change.

### PR DESCRIPTION
**Pull request by Anupam-dagar**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Currently the sales end date of tickets doesn't change when the start date of the event is changed to a time earlier than the current time.

#### Changes proposed in this pull request:

- Add an action which makes the sales end date to the event start date when the event is preponed.

Fix/es: #2649

### CODE REVIEW

**Comment by Anupam-dagar** [comment]
my bad, I meant the current set time

File: app/components/forms/wizard/basic-details-step.js [hunk 1]
```diff
@@ -346,6 +346,14 @@ export default Component.extend(FormMixin, EventWizardMixin, {
       }));
     },
 
+    salesEndDate(eventStartDate) {
```
Comment (CosmicCoder96) [inline app/components/forms/wizard/basic-details-step.js]: name this action as updateSalesEndDate

File: app/components/forms/wizard/basic-details-step.js [hunk 2]
```diff
@@ -346,6 +346,14 @@ export default Component.extend(FormMixin, EventWizardMixin, {
...
+      this.get('data.event.tickets').forEach(ticket => {
+        if (ticket.get('salesEndsAt').format('MM/DD/YYYY') > eventStartDate) {
```
Comment (CosmicCoder96) [inline app/components/forms/wizard/basic-details-step.js]: why do we need to format the date for comparing. Standard moment objects should be used for any comparisons.

Comment (Anupam-dagar) [reply app/components/forms/wizard/basic-details-step.js]: The input coming from the datepicker is a string and hence the format is used. Should I convert the string to moment object?

Comment (CosmicCoder96) [reply app/components/forms/wizard/basic-details-step.js]: @Anupam-dagar Yes.

**Comment by Anupam-dagar** [comment]
@CosmicCoder96 Please review.

**Pull request merged**